books/tests.py

```diff
@@ -0,0 +1,84 @@
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from .models import book
+
+
+class BookTests(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        testuser1 = get_user_model().objects.create_user(
+            username="testuser1", password="pass"
+        )
+        testuser1.save()
+
+        test_book = book.objects.create(
+            name="Harry Potter",
+            owner=testuser1,
+            publication_date="2022-05-09",
+            description="Magically Magic Stuff.",
+        )
+        test_book.save()
+
```

Comment:
If the tests are erroring out because of some user related error message, may also need to add a setUp method for user login. eg:

```
    def setUp(self):
        self.client.login(username="testuser1", password="pass")
```